### DISCUSSION TITLE
Hot reload fails in VS Code

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#18078] [Q&A]
HR fails In 2 different computers one with VS Code and the other VS Code insiders.

It works using Visual Studio but it fails in VS Code. (I tried reloading window)

VS 17.12.0 Preview 1.0

![image](https://github.com/user-attachments/assets/39de65d8-2386-4875-a41c-4a78ac722bdf)

![image](https://github.com/user-attachments/assets/dd14f0a0-3156-4970-8a96-3ef0775799ff)

This is the Uno hot reload output:

[hot-reload.txt](https://github.com/user-attachments/files/16814399/hot-reload.txt)

And this is the sample I've used.

[testHotReload.zip](https://github.com/user-attachments/files/16814410/testHotReload.zip)

**Comment by MartinZikmund** [comment]
@dr1rrb

**Comment by dr1rrb** [comment]
Realized that there is something that is really error prompt which can cause such connection errors, when we change the TFM, we need to make sure to launch profile in the "Run and Debug" option. Also has the TFM is reset when we close and re-open VS code, **this has to be validated when we re-open a project**.

E.g.
![image](https://github.com/user-attachments/assets/facd98f6-3481-4828-a65c-2d3f2f8f7798)

Also, as are using an external agent to detect and push changes in code to the running app (a.k.a. the "dev-server"), the debugger **must not** be attached to the app. This means that we cannot use the "play" button in the "Run and Debug" pane, but instead we need to use the menu "Run / Run Without Debugging" or press `Ctrl+F5`:

![image](https://github.com/user-attachments/assets/49da4dac-e268-4442-836a-e62d47208278)

**Reply by dr1rrb** [reply]
Also, if you don't see the "net8.0-browserwasm" indicator at the bottom, it's probably because there no active project selected (it's possible that you are is still starting even if not properly selected!). You need to click the folder name and select the MyProjectName**.csproj**:

![image](https://github.com/user-attachments/assets/4ae5c043-2dc2-480c-90e4-9a95aca7db7b)

**Comment by mcNets** [comment]
@dr1rrb, 
 
I've created a new solution selecting desktop TFM only.

I see that right now Sdk has changed to 5.3.108, this morning it was 5.3.99.

With 108 I can't even run the project, progress ring on 'Projects' is running for ever.

![image](https://github.com/user-attachments/assets/b45967d4-afee-4c00-8313-a16df1cc354e)

If I try to run the project using Ctrl+F5, I get:
```
fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint [fe80::eb41:4de1:a9a:7c7e].
fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint 172.22.224.1.
fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint [fe80::dd88:447c:7f84:c36d].
fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint 192.168.1.150.
fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint [fe80::851a:29c1:b786:dc63].
fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint 169.254.45.158.
fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint [fe80::960e:904c:f570:e29d].
fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint 169.254.24.13.
fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint [fe80::9f76:7169:e99a:ade2].
fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint 169.254.49.129.
fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get any remote control server endpoint from the IDE.
```

So I moved to 99 again and used Ctrl + F5, and HR it's working.

Sorry, my fault. I was using F5 to debug.

**Reply by dr1rrb** [reply]
Great to hear that 99 is working, but if 108 is broken it's problematic. Maybe @spouliot have some insight on this?

**Reply by agneszitte** [reply]
@kazo0 opened an issue for this https://github.com/unoplatform/uno/issues/18139
